astrocut/cutout_processing.py

```diff
@@ -0,0 +1,429 @@
+    
+    footprints = []
+    for bounds in combined_bounds:
+        footprints.append(_get_args(bounds, img_wcs))
+        
+    return footprints 
+
+
+def _moving_target_focus(path, size, cutout_fles, verbose=False):
+    """
+    Given a moving target path (list of RA/Decs) and size, that intersects with 
+    the given cutout(s) make a cutout of requested size centered on the 
+    moving target given by the path.
+    
+    Note: No resampling is done so there will be some jitter in the moving target
+    placement.
+    
+    Parameters
+    ----------
+    path : `~astropy.table.Table`
+        Table (or similar) object with columns "time," containing `~astropy.time.Time`
+        objects and "position," containing `~astropy.coordinate.Skycoord` objects.
+    size : array
+        Size in pixels of the cutout rectangle centered on the path locations ther will be 
+        returned. Formatted as [ny,nx]
+    cutout_fles : list
+        List of strings that are Target Pixel File paths.
+    verbose : bool
+        Optional. If true intermediate information is printed.
+        
+    Returns
+    -------
+    response : `~astropy.table.Table`
+        New cutout table.
+    """
+    
+    cutout_table_list = list()
+    
+    for fle in cutout_fles:
+        if verbose:
+            print(fle)
+        
+        # Get the stuff we need from the cutout file
+        hdu = fits.open(fle)
+        cutout_table = Table(hdu[1].data)
+        cutout_wcs = WCS(hdu[2].header)
+        hdu.close()
+        
+        
+        path["x"], path["y"] = cutout_wcs.world_to_pixel(path["position"])
+        # This line might need to be refined
+        rel_pts = ((path["x"] - size[0]/2 >= 0) & (path["x"] + size[0]/2 < cutout_wcs.array_shape[1]) & 
+                   (path["y"] - size[1]/2 >= 0) & (path["y"] + size[1]/2 < cutout_wcs.array_shape[0]))
+        
+        tck_tuple, u = splprep([path["x"][rel_pts], path["y"][rel_pts]], u=path["time"][rel_pts].jd, s=0)
+        
+        
+        cutout_table["time_jd"] = cutout_table["TIME"] + 2457000  # TESS specific code
+        cutout_table = cutout_table[(cutout_table["time_jd"] >= np.min(path["time"][rel_pts].jd)) & 
+                                    (cutout_table["time_jd"] <= np.max(path["time"][rel_pts].jd))]
+        
+        
+        cutout_table["x"], cutout_table["y"] = splev(cutout_table["time_jd"], tck_tuple)
+        cutout_table["bounds"] = _get_bounds(cutout_table["x"], cutout_table["y"], size)
+        
+        
+        cutout_table["TGT_X"] = cutout_table["x"] - cutout_table["bounds"][:, 0, 0]
+        cutout_table["TGT_Y"] = cutout_table["y"] - cutout_table["bounds"][:, 1, 0]
+        
+        positions = cutout_wcs.pixel_to_world(cutout_table["x"], cutout_table["y"])
+        cutout_table["TGT_RA"] = positions.ra.value
+        cutout_table["TGT_DEC"] = positions.dec.value
+        
+        cutout_table["bounds"] = [(slice(*y), slice(*x)) for x, y in cutout_table["bounds"]]
```

Comment:
In most cases you have bounds being [[x],[y]]. I want to make sure here you do mean to have `slice(*y)` first. 

Some extra comments around this for loop could also help.

Reply:
Yeah, this is correct, I think it's because of the difference in x,y vs y,x in fits vs python. 😛 I'll add some more comments.